base/src/main/java/org/geysermc/api/util/ApiVersion.java

```diff
@@ -0,0 +1,60 @@
+package org.geysermc.api.util;
+
+/**
+ * Represents a version of the api.
+ */
+public class ApiVersion {
+    private final int major;
+    private final int minor;
+    private final int patch;
+
+    public ApiVersion(int major, int minor, int patch) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+    }
+
+    /**
+     * Returns the major version of the api.
+     *
+     * @return the major version
+     */
+    public int major() {
+        return this.major;
+    }
+
+    /**
+     * Returns the minor version of the api.
+     *
+     * @return the minor version
+     */
+    public int minor() {
+        return this.minor;
+    }
+
+    /**
+     * Returns the patch version of the api.
+     *
+     * @return the patch version
+     */
+    public int patch() {
+        return this.patch;
+    }
+
+    /**
+     * Checks whether the given version is compatible with this version.
+     * The parameters show the desired API version, which is checked against this present API version.
+     * The major versions must match, and the desired minor version must be equal or smaller than this minor version.
+     * If the minor versions match, then the desired patch version must be equal or smaller than this patch version.
+     *
+     * @param major the desired major version
+     * @param minor the desired minor version
+     * @param patch the desired patch version
+     * @return true if the desired API version is compatible with the present API version
+     */
+    public boolean isCompatible(int major, int minor, int patch) {
```

Comment:
I think that `isCompatible` says too little about the actual comparison that's being made.
Maybe we could rename it to `isOlderThan` or `isNotOlderThan` or have one method for each of them.
And I would probably replace the arguments with an `ApiVersion`

Reply:
I wouldnt split it into two methods since we only need to know if the api requested is the same version or older.. both isOlderThan and isNotOlderThan imply that only older versions are supported...
Personally, isCompatible is enough here, since the javadocs give additional info as to how compatibility works here. If a rename is better, i would opt for ```isOlderOrEqual``` and keep the one method

Reply:
Regarding using an ApiVersion as an argument - i initially was doing that, until Konica advised against it since it's pointless to create a wrapper - aka https://github.com/geyserMC/api/pull/1#discussion_r1232357919

Reply:
As noted in the Geyser PR I don't think we actually need this method, but if it ends up staying I'd use ApiVersion as argument as it's easier to use and has no impact on performance.

Reply:
I think that having this kind of method will help us standardize what versioning scheme this class is actually meant to represent. Its a programmatic way of enforcing the same compatibility rules across our multiple APIs.

As a bonus, it exposes the compatibility rules to the API. If an external plugin/mod wants to check the Base or Geyser API version before making more complex calls, they would have to implement their own comparison code. That would be less likely to happen, and is also more prone to inaccuracy.

Side note: if necessary, an implementation could override this method to expose certain edge cases that stray from the standard for whatever reason. For example, the workaround in https://github.com/GeyserMC/Geyser/pull/3880.

The reason I suggested not having ApiVersion as the parameter type was because at the time, the method wrongly appeared commutative both by signature and javadoc. The int parameters also discourage the use of ApiVersion by API consumers, which I think is a good thing considering the presence of this method in the class. This method could be moved elsewhere, but I don't think that's the best option for reasons I won't get into now.

In regards to method names, I don't think the suggestions with "older" in the name are that clear to be honest. The requested API version is allowed to be older, but only up to a certain point. Things like `supportsConsumerTarget` or `supportsRequestedVersion` are verbose but imo communicate the relationship clearly. 

Anyway, I didn't mean to be so wordy, but here we are lol

Reply:
I've implemented an approach to hopefully address these (thanks konica for the ideas!); how does the current iteration look to you?